### PR TITLE
Feature/stack

### PR DESCRIPTION
**Pull request by UnderTheKnife**

### CODE REVIEW

File: Stack/js/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/**
+ *
+ * @param {object} parent
+ * @returns {Element}
+ */
+function createUl(parent) {
+  let ul = window.document.createElement('ul');
+  let line = window.document.createElement('hr');
+
+  line.style.margin = '5px';
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: Set CSS style in CSS file. Don't use JS for it.

File: Stack/js/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  parent.appendChild(ul);
+  parent.appendChild(line);
+
+  return ul;
+}
+
+/**
+ *
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: Please, add small comments for method. What it doing

File: Stack/js/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const values = window.document.getElementById('values');
+  let arr = [];
+
+  for (let el = 0; el < values.children.length; el += 1) {
+    if (values.children[el].value !== '') {
+      arr.push(values.children[el].value);
+    }
+  }
+
+  window[func](arr);
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: Try to call method like that:
```func(arr);```

File: Stack/js/index.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const button = window.document.getElementById('add');
+  let child = window.document.createElement('input');
+
+  values.insertBefore(child, button);
+}
+
+window.addEventListener('load', () => {
+  window.document.getElementById('add').addEventListener('click', add);
+  window.document.getElementById('stackButton').addEventListener('click', () => {
+    push('stack');
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: don't send string name of method to push. Try to send method name, like:
```push(stack);```

File: Stack/css/styles.css [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+.answer {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+  text-align: center;
+}
+
+.stack, .queue {
```
Comment (StasVysotskyi) [inline Stack/css/styles.css]: Try to keep your CSS file in one style. Between selectors you have to add 1 empty line. If you need to add 2 selectors for 1 styles, please keep in following style:
```
.stack,
.queue {
...
}
```

File: Stack/js/index.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  return ul;
+}
+
+/**
+ * Show stack
+ * @param arr
+ */
+function stack(arr) {
+  const parent = document.getElementById('stack');
+  let ul = createUl(parent);
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: I think ul variable can be const, is not?

File: Stack/js/index.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Create new 'ul' element
+ * @param {object} parent
+ * @returns {Element}
+ */
+function createUl(parent) {
+  let ul = document.createElement('ul');
+  let line = document.createElement('hr');
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: I think ul and line variables can be const, is not?

File: Stack/js/index.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+ * Show stack
+ * @param arr
+ */
+function stack(arr) {
+  const parent = document.getElementById('stack');
+  let ul = createUl(parent);
+  let count = arr.length;
+  let li;
+
+  for (let id = 0; id < count; id += 1) {
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: If you need to use count variable only for loop, you can get length of arr in the loop, like:
```for (let id = 0; id < arr.length; id += 1) {```
And you don't need allocate memory for it

File: Stack/js/index.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    li = document.createElement('li');
+    li.innerHTML = arr.pop();
+    ul.appendChild(li);
+  }
+
+}
+
+/**
+ * Show queue
+ * @param arr
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: You also can decelerate param type in the JSDoc. 
```@param {Array} arr```

File: Stack/js/index.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+ */
+function queue(arr) {
+  const parent = document.getElementById('queue');
+  let ul = createUl(parent);
+  let count = arr.length;
+  let li;
+
+  for (let id = 0; id < count; id += 1) {
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: you can use simple version of array loop:

```
arr.forEach(() => {
  li = document.createElement('li');
  li.innerHTML = arr.shift();
  ul.appendChild(li);
});
```

How you can see, you don't need to setup loop and get count of array.  It's easier

File: Stack/js/index.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+ */
+function push(func) {
+  const values = document.getElementById('values');
+  let arr = [];
+
+  for (let el = 0; el < values.children.length; el += 1) {
+    if (values.children[el].value !== '') {
+      arr.push(values.children[el].value);
+    }
+  }
```
Comment (StasVysotskyi) [inline Stack/js/index.js:62]: Try this way:

```
values.children.forEach(child => {
  if (child.value !== '') {
    arr.push(child.value);
  }
});
```

File: Stack/js/index.js [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  func(arr);
+}
+
+/**
+ * Add new input
+ */
+function add() {
+  const values = document.getElementById('values');
+  const button = document.getElementById('add');
+  let child = document.createElement('input');
```
Comment (StasVysotskyi) [inline Stack/js/index.js]: Why child is variable? Why not const?

File: Stack/js/index.js (hunk 8 above)
Comment (UnderTheKnife) [reply Stack/js/index.js]: I can't, because the array changes its length

File: Stack/js/index.js (hunk 10 above)
Comment (UnderTheKnife) [reply Stack/js/index.js]: I can't, because the array changes its length

File: Stack/js/index.js (hunk 11 above)
Comment (UnderTheKnife) [reply Stack/js/index.js:62]: It's doesn't work.

Comment (StasVysotskyi) [reply Stack/js/index.js:62]: Yep, values.children in this context like Array object, but isn't real array. So you cannot use forEach

**Review by StasVysotskyi — APPROVED** [review]

**Pull request merged**